nnstreamer/tensor_trainer/tensor_trainer_nntrainer.cc

```diff
@@ -0,0 +1,471 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+/**
+ * NNStreamer tensor_trainer subplugin for nntrainer
+ * Copyright (C) 2022 Hyunil Park <hyunil46.park@samsung.com>
+ */
+/**
+ * @file   tensor_trainer_nntrainer.cc
+ * @date   02 Dec 2022
+ * @brief  NNStreamer tensor_trainer subplugin
+ * @see    http://github.com/nnstreamer/nnstreamer
+ * @author Hyunil Park <hyunil46.park@samsung.com>
+ * @bug    No known bugs except for NYI items
+ */
+#include "tensor_trainer_nntrainer.hh"
+#include <cstring>
+#include <iostream>
+#include <nntrainer_log.h>
+#include <pthread.h>
+#include <random>
+#include <sys/syscall.h>
+#include <thread>
+#include <unistd.h>
+
+#define UNUSED(expr) \
+  do {               \
+    (void)(expr);    \
+  } while (0)
+
+/**
+ * @brief startup constructor
+ */
+void init_subplugin_nntrainer(void) __attribute__((constructor));
+
+/**
+ * @brief startdown destructor
+ */
+void fini_subplugin_nntrainer(void) __attribute__((destructor));
+
+void nntrainer_thread_func(NNTrainer::NNTrainerTrain *nntrainer) {
+  nntrainer->trainModel();
+}
+
+static int nntrainer_model_invoke(const GstTensorTrainerFramework *fw,
+                                  const GstTensorTrainerProperties *prop,
+                                  void *private_data,
+                                  const GstTensorMemory *input) {
+  NNTrainer::InputTensorsInfo *data = nullptr;
+  NNTrainer::NNTrainerTrain *nntrainer =
+    reinterpret_cast<NNTrainer::NNTrainerTrain *>(private_data);
+  UNUSED(fw);
+  ml_logd("<called>");
+
+  if (!nntrainer) {
+    std::cerr << "Failed get nntrainer" << std::endl;
```

Comment:
https://www.geeksforgeeks.org/avoid-using-stdendl/

Reply:
Thank you. I will fix it

Reply:
Done, ([3080216](https://github.com/nnstreamer/nntrainer/pull/2059/commits/308021689f0e5dd68720d84fc986b6c222ead9e7)), Thank you.